### DISCUSSION TITLE
Welcome to exchangelib Discussions!

### DISCUSSION THREAD

**Discussion by ecederstrand** [ecederstrand/exchangelib#837] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about
  * Share ideas
  * Engage with other community members
  * Welcome others and are open-minded. Remember that this is a community we build together 💪

  To get started, comment below with an introduction of yourself and tell us about yourself and how you're using exchangelib.

**Comment by pkonnov** [comment]
:v:
Where can I keep track of changes in Outlook functionality?
10/26/2020 - when a one-time appointment was rescheduled, the calendar item was not recreated, now it is, yes?

```python
    def get_calendar_items(self) -> typing.List[CalendarItem]:
        sub_times = get_timedelta(minutes=10)
        items = self.account.calendar.filter(datetime_created__gte=self.now - sub_times). \
            order_by('-datetime_received')
        return items
```

and it comes again as a new element...

**Reply by ecederstrand** [reply]
I'm not sure what you are suggesting here. Can you elaborate? Is this a change in functionality in exchangelib? While Outlook is also a client of the EWS protocol, exchangelib does not rely on Outlook in any way.

**Reply by pkonnov** [reply]
Today I noticed that when rescheduling a meeting to another time, a new one is created, with the time to which it was rescheduled. In October 2020, this behavior was not observed.

**Reply by ecederstrand** [reply]
Are you rescheduling with exchangelib, or with Outlook?

**Reply by pkonnov** [reply]
Created with exchangelib, rescheduling with Outlook

**Reply by ecederstrand** [reply]
Ok. In that case I don't think it's something exchangelib can do anything about.

**Comment by kanns15** [comment]
ecederstrand/ALL, We are using excahngelib to classify the emails from shared mailbox, basically auto tag the emails with proper labels based on machine learning predicted values.
Below are the ideas we have to improvise the tool, wondering is it all possible in exchangelib, if so can u please refer me to those links,

1.  email response time (how much time passed from when the email was viewed to when the reply is sent)
2.  alerting/notifications for specific emails
3.  Move important emails to top of the inbox queue.

**Reply by ecederstrand** [reply]
1. To my knowledge, it's not possible to know when a message was viewed, unless the sender requested read receipts and the client is configured to send them. The `datetime_received` field tells you when the email was received on the server, but that doesn't seem like what you want.
2. You can set that up already. Just create a filter that covers the emails you want to get notifications for, and run that filter regularly.
3. An inbox has no natural order of messages on the server. Outlook will order messages by received date descending, but that's a client-side setting. You can tell EWS to sort messages, and exchangelib allows you to sort by (almost) any field.

**Comment by Gil-nuriel** [comment]


**Comment by pkonnov** [comment]
:v: 
When you move the meeting days in the meeting cycle, a new meeting cycle is created. A letter arrives with the transfer of the meeting and a letter with a newly created meeting. Is that how it should be?

**Reply by ecederstrand** [reply]
Is it what you expected? Is it different from how other clients handle changed dates?

**Reply by pkonnov** [reply]
I assumed that the meeting days for this cycle of meetings would change. Without deleting the current cycle.

**Reply by pkonnov** [reply]
Is it possible to get from the letter with a new meeting, an indication that this meeting was created from another meeting cycle?

**Reply by ecederstrand** [reply]
Are you editing the master recurrence or a single occurrence? The `send_meeting_invitations` argument for `CalendarItem.save()` allows customizing how meeting updates are announced, but otherwise I think this is handled within the Exchange server.

How does this behave if you update the meeting from e.g. Outlook or OWA?

**Reply by pkonnov** [reply]
I am editing a single occurrence in an appointment loop.
Outlook.

1. A cycle of meetings is created. Every Wed, Fri for 2 months from 14-00 to 14-30.
2. Meetings is 24/26 - February, 3/5 10/12 17/19 ... - March ...
3. We click on March 17th. Click "Edit -> This and all following events"
4. Opens to the window with a meeting
5. We change the days of the event. For example, we add holding meetings on Thursdays.
6. Now we have meetings starting from March 17th on Tue Wed Thu
7. Click send.
8. A new meeting cycle is created from this meeting cycle.

I think this is the way it should be. :+1:

**Reply by ecederstrand** [reply]
Ok, so it seems that Outlook changes the end date of the existing master recurrence and creates a new one for the remaining period. You could inspect the calendar items with exchangelib when doing the above, to verify that this is in fact what happens. If so, you could do the same in exchangelib. I'm not sure it's worthwhile to add the "Edit -> This and all following events" functionality to exchangelib - the building blocks are there, at least.

**Comment by Gil-nuriel** [comment]
It is possible to search for emails in all the inboxes I have aceess to? 
e.g instead to specify primary_smtp_adress to search for a specific email with the subject "Test", I want to search all the emails with the subject "Tests" from ALL the inboxes of the organization

**Reply by ecederstrand** [reply]
No. That is not supported by EWS.

**Reply by Gil-nuriel** [reply]
Just to clarify the scenario I want to implement.
I want to delete specific email by sender/subject/email-id from ALL the mailboxes that match my query.
right now I'm able to do this action only from one user which I specify in the Account function as "primary_smtp_adress"

**Reply by Gil-nuriel** [reply]
Got it, thanks for the fast reply

**Reply by ecederstrand** [reply]
To delete items, you use the DeleteItem service: https://docs.microsoft.com/en-us/exchange/client-developer/web-service-reference/deleteitem-operation It only supports deleting Item IDs from one mailbox at a time.

**Reply by Gil-nuriel** [reply]
there's a way to get a list of ALL the users I have access to?

**Reply by ecederstrand** [reply]
I don't think EWS supports this, unfortunately.
